app/controllers/application.js

```diff
@@ -49,6 +49,18 @@ export default Controller.extend(cancelOrderMixin, {
   cartscroll: service(),
 
   hasCartItems: alias("cart.isNotEmpty"),
+
+  hasValidCartItems: computed(
+    "cart.unavailableCartItems",
+    "cart.isNotEmpty",
+    function() {
+      return (
+        this.get("cart.isNotEmpty") &&
+        this.get("unavailableCartItems.length") === 0
```

Comment:
I think we don't need to add equality check here,  just use of length is enough. 0 is considered false.

`var a = ""
a.length ? "a" : "b"
=> "b"
var a = "hello"
a.length ? "a" : "b"
=> "a"`

Reply:
Since using this value for `disabled` attribute in the handlebar template, when evaluated to `0` it sets the value as `disabled="0"` and it just disables the button. Hence passed explicit check which will either return a true or false value.

Reply:
Okay. Sounds Good :+1: :)